query/physicalplan/aggregate.go

```diff
@@ -236,103 +238,110 @@ func hashInt64Array(arr *array.Int64) []uint64 {
 	return res
 }
 
-func (a *HashAggregate) Callback(ctx context.Context, r arrow.Record) error {
-	// Generates high volume of spans. Comment out if needed during development.
-	// ctx, span := a.tracer.Start(ctx, "HashAggregate/Callback")
-	// defer span.End()
-
-	groupByFields := a.groupByFields
-	groupByFieldHashes := a.groupByFieldHashes
-	groupByArrays := a.groupByArrays
-
-	defer func() {
-		groupByFields = groupByFields[:0]
-		groupByFieldHashes = groupByFieldHashes[:0]
-		groupByArrays = groupByArrays[:0]
-	}()
-
-	var columnToAggregate arrow.Array
-	aggregateFieldFound := false
-
-	for i, field := range r.Schema().Fields() {
-		for _, matcher := range a.groupByColumnMatchers {
-			if matcher.MatchColumn(field.Name) {
-				groupByFields = append(groupByFields, field)
-				groupByFieldHashes = append(groupByFieldHashes, scalar.Hash(a.hashSeed, scalar.NewStringScalar(field.Name)))
-				groupByArrays = append(groupByArrays, r.Column(i))
-			}
-		}
+func (a *HashAggregate) SetNext(next PhysicalPlan) {
+	a.next = next
 
-		if a.columnToAggregate.MatchColumn(field.Name) {
-			columnToAggregate = r.Column(i)
-			aggregateFieldFound = true
-		}
-	}
+	a.callbacks = make([]logicalplan.Callback, 0, len(next.Callbacks()))
+	for range next.Callbacks() {
```

Comment:
To be clear: is your idea to merge callback awareness in operators as a first step and then move this to the physical plan stage? I'm very uncomfortable with having planning code in execution but ok if it's temporary